joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/AstCreationPass.scala

```diff
@@ -1,19 +1,18 @@
 package io.joern.rubysrc2cpg.passes
 import io.joern.rubysrc2cpg.astcreation.AstCreator
+import io.joern.rubysrc2cpg.utils.{PackageContext, PackageTable}
 import io.joern.x2cpg.SourceFiles
 import io.joern.x2cpg.datastructures.Global
-import io.joern.x2cpg.passes.frontend.TypeNodePass
 import io.shiftleft.codepropertygraph.Cpg
 import io.shiftleft.passes.ConcurrentWriterCpgPass
 import org.slf4j.LoggerFactory
 
-import java.util.concurrent.ConcurrentHashMap
 import scala.jdk.CollectionConverters.EnumerationHasAsScala
 
-class AstCreationPass(inputPath: String, cpg: Cpg) extends ConcurrentWriterCpgPass[String](cpg) {
+class AstCreationPass(inputPath: String, cpg: Cpg, global: Global, packageTable: PackageTable, tempDir: String)
```

Comment:
Please pass a flat here instead of string that determines whether the AST should be applied to the diffgraph